app/src/main/java/com/fastlink/zekrapp/presentation/ZekirScreen.kt

```diff
@@ -35,77 +34,58 @@ import androidx.compose.ui.res.stringResource
-import kotlinx.coroutines.delay
-import kotlinx.coroutines.launch
-
+import com.fastlink.zekrapp.presentation.utils.halfCircle
 
 
 @Composable
 fun ZekirScreen(categoryId: Int) {
-    val toast = LocalToast.current
+//    val toast = LocalToast.current
     val clipboardManager = LocalClipboardManager.current
     val viewModel = LocalViewModel.current
     val scope = rememberCoroutineScope()
-    val category = viewModel.getCategoryById(categoryId)
-    val zekirs = viewModel.getZekirByCategoryId(category.id)
-    val zekirNumber = remember {
-        mutableIntStateOf(0)
-    }
-    val zekirCounter = remember {
-        mutableIntStateOf(0)
-    }
-    LaunchedEffect(true) {
-        toast(" الذكر ${zekirNumber.intValue + 1} من ${zekirs.size}")
+    val category = remember {
+        viewModel.getCategoryById(categoryId)
     }
+
+        viewModel.getZekirByCategoryId(category.id)
+
+
+    val zekirNumber = viewModel.zekirNumber
+    val zekirCounter = viewModel.zekirCounter
+
+//    LaunchedEffect(true) {
+//        toast(" الذكر ${zekirNumber.intValue + 1} من ${zekirs.size}")
+//    }
     val animatedBorder by animateFloatAsState(
         targetValue =
-        (zekirCounter.intValue.toFloat() / zekirs[zekirNumber.intValue].counterNumber.toFloat()) * 360f,
+        (zekirCounter.intValue.toFloat() / viewModel.zekirs[zekirNumber.intValue].counterNumber.toFloat()) * 360f,
         label = "",
         animationSpec = tween(1000)
     )
 
-    fun onClicked() {
-        if (zekirCounter.intValue < zekirs[zekirNumber.intValue].counterNumber) {
-            zekirCounter.intValue++
-            if (zekirCounter.intValue == zekirs[zekirNumber.intValue].counterNumber) {
-                scope.launch {
-                    delay(1500)
-                    if (zekirNumber.intValue < zekirs.size - 1) {
-                        zekirCounter.intValue = 0
-                        zekirNumber.intValue++
-                        scope.launch {
-                            toast(" الذكر ${zekirNumber.intValue + 1} من ${zekirs.size}")
-                        }
-                    }
-                }
-            }
-        }
-    }
 
     Scaffold(
         backgroundColor = MaterialTheme.colorScheme.background,
         topBar = {
-            ZekirScreenAppBar(title = category.categoryTitle)
+            ZekirScreenAppBar(title =category.categoryTitle)
         },
         floatingActionButton = {
             Box(
                 modifier = Modifier
                     .size(90.dp)
-                    .clip(CircleShape)
-                    .background(MaterialTheme.colorScheme.background),
+                    .halfCircle(MaterialTheme.colorScheme.background),
                 contentAlignment = Alignment.Center,
             ) {
                 FloatingActionButton(
                     onClick = {
-                        onClicked()
+                        viewModel.onClicked(scope)
```

Comment:
As I mentioned before, onClicked is not enough, because here you have a lot of components, you should mention what on what clicked.